packages/typespec-client-generator-core/src/types.ts

```diff
@@ -1109,15 +1198,19 @@ function addPropertiesToModelType(
   operation?: Operation
 ): [void, readonly Diagnostic[]] {
   const diagnostics = createDiagnosticCollector();
-  for (const property of type.properties.values()) {
+  const httpOpParts = getHttpOperationParts(context, operation);
```

Comment:
can we put the entries of adding multi-part info into http operation handling logic in http.ts? i'm trying to not to add too much protocol special logic in the type handling process (some refactoring work for tcgc). if a request is multi-part, then add new info to sdkbodymodelpropertytype.

Reply:
According to our design https://github.com/Azure/typespec-azure/pull/987, multipart info is bound with property of model instead of operation level. As declared in https://github.com/Azure/typespec-azure/pull/1090#discussion_r1682069461, if no special logic in getClientType, TCGC will produce some unexpected sdkType named `HttpPart`.

Reply:
emmm, is it possible to remove the httppart wrapper when handling model property since it is kind of special body model property instead of remove the wrapper in get client type?

Reply:
After offline discussion with @tadelesh, `getClientType` will not be used for type of ModelProperty so I will try to centralize the logic in one place.